Sources/EdgeNetworkHandlers/EdgeHitProcessor.swift

```diff
@@ -120,8 +127,11 @@ class EdgeHitProcessor: HitProcessing {
         networkService.doRequest(url: url,
                                  requestBody: edgeHit.request,
                                  requestHeaders: headers,
-                                 responseCallback: callback,
-                                 completion: completion)
+                                 responseCallback: callback) { [weak self] success, retryInterval in
+            // remove any retry interval if success or has default interval value, add retry interval if failed
+            self?.entityRetryIntervalMapping[entityId] = success || retryInterval == EdgeConstants.Defaults.RETRY_INTERVAL ? nil : retryInterval
```

Comment:
We should explicitly call the removeValue API to remove the pre-existing mapping in case the internal implementation of ThreadSafeDictionary changes in the future. Considering TimeInterval is optional, you can have something like this:
```
if success {
     // remove retry interval if success
     _ = self?.entityRetryIntervalMapping.removeValue(forKey: entityId)
} else if retryInterval != nil {
     // add retry interval if failed
     self?.entityRetryIntervalMapping[entityId] = retryInterval
}
...
```

Reply:
The existing line of code already does this. Setting a value in a dictionary to nil is the equivalent to removing it explicitly. We would never modify the implementation of ThreadSafeDIctionary to make it not behave like a standard Swift dict when used through the subscript.

Reply:
Since it is out of this extension's control I would not rely that there will be no changes. The APIs can remain the same and the underlying data structure can be different. Otherwise why do we have a removeValue API?

Reply:
It actually is somewhat in control of the extension based on the definition of `ThreadSafeDictionary<String, TimeInterval>()` 

The definition explicitly states it cannot hold `nil` values, thus setting any key to `nil` can have no other side-effect than to remove it from the dictionary if present.